horovod/common/ops/nccl_operations.cc

```diff
@@ -395,5 +395,116 @@ Status NCCLBroadcast::Execute(std::vector<TensorTableEntry>& entries,
   return gpu_op_context_.FinalizeGPUQueue(entries);
 }
 
+Status NCCLAllgather::Execute(std::vector<TensorTableEntry>& entries,
+                                const Response& response) {
+  auto& first_entry = entries[0];
+
+  gpu_op_context_.InitGPU(entries);
+  nccl_op_context_.InitNCCLComm(entries, response.devices());
+  gpu_op_context_.InitGPUQueue(entries, response);
+
+  // Sizes of subcomponents of each entry from all ranks
+  auto** entry_component_sizes = new int64_t* [entries.size()];
+
+  // Offset of each subcomponent of every entry in the final buffer after
+  // allgatherv
+  auto** entry_component_offsets = new int64_t* [entries.size()];
+
+  int global_size = global_state_->controller->GetSize();
+  int global_rank = global_state_->controller->GetRank();
+  auto* recvcounts = new int[global_size]();
```

Comment:
The allocations with new are not being deleted in this routine, but need to be to avoid memory leaks. This was fixed recently for the other allgather implementations in #1845.

Reply:
I have fixed it.